### PR TITLE
4.D: Adding Thunk middleware to call a mock list of conferences

### PR DESCRIPTION
**Pull request by resong**
Trello Number: 4.D

Task: Add code to use thunk middleware to make an API call to display a list of conferences

What does this code do?
Uses thunk middleware to make an API call to display a list of conferences

Do I need to yarn install?
no

Success criteria:
Project doesn't explode and you can see a list of events on the main page after running `yarn start`: 
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -11,30 +12,77 @@ import PageTitle from "./components/PageTitle.js";
...
+        }
       </div>
     );
   }
 }
 
-export default App;
+// Mapping state and dispatch
+// to props and connecting them
+// --> To be extracted out into separate containers file?
```
Comment (pkanal) [inline src/App.js]: Let's do this in this PR? Its usually better to separate out components early.

File: src/api/conferences.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import conferences from './mockData.json';
+
+/**
+ * Mocking call to API to get list of conferences
+ */
+
+// Timeout in ms
+const TIMEOUT = 1000;
+
```
Comment (pkanal) [inline src/api/conferences.js:9]: Can we leave a TODO here to swap out for the real api call?

File: src/store/createStore.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import { rootReducer } from '../reducers/rootReducer';
+
+/**
+ * Redux store creation, incorporating redux-thunk
+ */
+export const store = createStore(
+    rootReducer,
+    undefined,
+    applyMiddleware(thunk)
```
Comment (pkanal) [inline src/store/createStore.js]: Can we add logger middleware so that we can debug actions easily?

File: src/reducers/conferenceReducer.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { CONFERENCE_ACTIONS } from '../actions';
+import INITIAL_STATE from './initialState';
+
+/**
+ * Reducer for handling 
+ * actions associated with conferences
+ */
+export const conferenceReducer = (state = INITIAL_STATE, action) => {
```
Comment (pkanal) [inline src/reducers/conferenceReducer.js]: can we separate this into two reducers? `conferences` and `isLoading`?

File: src/reducers/rootReducer.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { combineReducers } from '../../../../../../../AppData/Local/Microsoft/TypeScript/2.9/node_modules/redux';
```
Comment (hellomkreyes) [inline src/reducers/rootReducer.js]: Della is coming over to talk to you re: this path @resong

File: src/api/conferences.js (hunk 2 above)
Comment (pkanal) [reply src/api/conferences.js:9]: This is done :D

**Comment by pkanal** [comment]
- Added redux logger
- Separated out events container
- Removed app test because its an integration test, todo: add unit tests for relevant containers

**Review by pkanal — APPROVED** [review]

**Pull request merged**